components/brave_wallet/browser/eth_json_rpc_controller.cc

```diff
@@ -133,28 +130,43 @@ void EthJsonRpcController::SetNetwork(Network network) {
                          subdomain.c_str(), GetInfuraProjectID().c_str());
   network_url_ = GURL(spec);
 
-  observers_->Notify(FROM_HERE,
-                     &BraveWalletProviderEventsObserver::ChainChangedEvent,
-                     GetChainIDFromNetwork(network_));
+  for (const auto& observer : observers_) {
+    observer->ChainChangedEvent(GetChainIdFromNetwork(network_));
```

Comment:
this won't get called for type `Localhost` or `Custom`

Reply:
Nice find, unrelated to this PR though so I posted this issue for it: https://github.com/brave/brave-browser/issues/17201

Reply:
Fixed after all since I need to run this through CI again anyway:
https://github.com/brave/brave-core/pull/9564/commits/56f6a3db242a3ce3554abfae6936fc719245a14e